config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -32,59 +41,147 @@
     private final Set<EntityBuilder> entityBuilders;
     private final BundleDocumentBuilder bundleDocumentBuilder;
     private final BundleMetadataBuilder bundleMetadataBuilder;
+    private final DocumentTools documentTools;
 
     @Inject
     BundleEntityBuilder(final Set<EntityBuilder> entityBuilders, final BundleDocumentBuilder bundleDocumentBuilder,
-                        final BundleMetadataBuilder bundleMetadataBuilder) {
+                        final BundleMetadataBuilder bundleMetadataBuilder, final DocumentTools documentTools) {
         // treeset is needed here to sort the builders in the proper order to get a correct bundle builded
         // Ordering is necessary for the bundle, for the gateway to load it properly.
         this.entityBuilders = unmodifiableSet(new TreeSet<>(entityBuilders));
         this.bundleDocumentBuilder = bundleDocumentBuilder;
         this.bundleMetadataBuilder = bundleMetadataBuilder;
+        this.documentTools = documentTools;
     }
 
-    public Map<String, Pair<Element, BundleMetadata>> build(Bundle bundle, EntityBuilder.BundleType bundleType,
-                                                            Document document, String projectName,
-                                                            String projectGroupName, String projectVersion) {
+    public Pair<Element, BundleMetadata> build(Bundle bundle, EntityBuilder.BundleType bundleType,
+                                               Document document, String projectName,
+                                               String projectGroupName, String projectVersion, AnnotatedEntity annotatedEntity) {
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
+        if(annotatedEntity != null){
+            if (EntityBuilder.BundleType.DEPLOYMENT == bundleType) {
+
+                List<Entity> entityList = getEntityDependencies(annotatedEntity.getPolicyName(), entities, bundle);
+                LOGGER.log(Level.FINE, "Annotated entity list : " + entityList);
+                Optional<Entity> rootEntity = entityList.stream().filter(entity -> annotatedEntity.getEntityName().equals(entity.getName())).findFirst();
+                if(!rootEntity.isPresent()){
```

Comment:
I feel finding root element is not required here, as it will be part of dependency map.
Even if you feel this is required, comparing only with name might give wrong entity, you should compare the type as well.

Reply:
root entity was missing when i was testing so added this

Reply:
added type check also